app/models/manageiq/providers/physical_infra_manager.rb

```diff
@@ -48,5 +48,32 @@ def console_supported?
     def console_url
       raise MiqException::Error, _("Console not supported")
     end
+
+    supports :change_password
+
+    # Changes the password of userId on provider client and database.
+    #
+    # @param [current_password] password currently used for connected userId in provider client
+    # @param [new_password]     password that will replace the current one
+    #
+    # @return [Boolean] true if the password was changed successfully
+    def change_password(current_password, new_password)
+      change_password_on_provider(current_password, new_password)
+      update_authentication(:default => {:userid => authentication_userid, :password => new_password})
```

Comment:
Instead of assuming `:default` you should take the auth_type as an argument, it can default to `:default` but should be overridable